packages/iTwinUI-react/src/core/Tile/Tile.tsx

```diff
@@ -141,6 +160,29 @@ export const Tile = (props: TileProps) => {
   const showMenu = React.useCallback(() => setIsMenuVisible(true), []);
   const hideMenu = React.useCallback(() => setIsMenuVisible(false), []);
 
+  const renderTitleIcon = () => {
```

Comment:
This can be moved into a private component at the bottom of the file, and then we can just use it in the JSX markup like this: `<TitleIcon status={status} isSelected={isSelected} isLoading={isLoading} isNew={isNew} />`.

The benefit is that react will not re-create this function on every render.